### PR TITLE
Extend test_syntax to check that canonicalization doesn't panic on the input

### PR DESCRIPTION
**Pull request by joshuawarner32**
Importantly, this would have caught the issue discovered in #7279, prior to that landing on main.

### CODE REVIEW

File: crates/compiler/test_syntax/src/test_helpers.rs [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
-    ast::{Defs, Expr, FullAst, Header, Malformed, SpacesBefore},
-    header::parse_module_defs,
-    normalize::Normalize,
-    parser::{Parser, SyntaxError},
-    state::State,
-    test_helpers::{parse_defs_with, parse_expr_with, parse_header_with},
+    ast::{Defs, Expr, FullAst, Header, SpacesBefore},
+    test_helpers::{parse_defs_with, parse_header_with},
 };
+// use roc_problem::can::Problem;
```
Comment (lukewilliamboswell) [inline crates/compiler/test_syntax/src/test_helpers.rs:21]: stray comment

File: crates/compiler/test_syntax/src/test_helpers.rs [hunk 2]
```diff
@@ -116,6 +133,81 @@ impl<'a> Output<'a> {
...
         }
     }
+
+    pub fn canonicalize(&self, arena: &Bump, src: &str) {
+        set_panic_not_exit(true); // can has a bunch of internal_error! calls
+
+        match self {
+            Output::Header(_) => {}
+            Output::ModuleDefs(_) => {
+                // TODO: canonicalize module defs
```
Comment (lukewilliamboswell) [inline crates/compiler/test_syntax/src/test_helpers.rs:143]: TODO

File: crates/compiler/test_syntax/src/test_helpers.rs [hunk 3]
```diff
@@ -116,6 +133,81 @@ impl<'a> Output<'a> {
...
+            }
+            Output::Full(_) => {
+                // TODO: canonicalize full ast
```
Comment (lukewilliamboswell) [inline crates/compiler/test_syntax/src/test_helpers.rs:146]: TODO

File: crates/compiler/test_syntax/tests/test_snapshots.rs [hunk 4]
```diff
@@ -698,6 +698,39 @@ mod test_snapshots {
         // END SNAPSHOTS (for automatic test detection via `env ROC_SNAPSHOT_TEST_OVERWRITE=1 cargo test`)
     }
 
+    /// Does the given test name expect the canonicalization process to panic?
+    fn expect_canonicalize_panics(test_name: &str) -> bool {
```
Comment (lukewilliamboswell) [inline crates/compiler/test_syntax/tests/test_snapshots.rs:702]: This looks helpful!

**Review by lukewilliamboswell — DISMISSED** [review]

File: crates/compiler/test_syntax/src/test_helpers.rs (hunk 2 above)
Comment (joshuawarner32) [reply crates/compiler/test_syntax/src/test_helpers.rs:143]: Yep; my plan is to leave this for later. Expr's are the majority of our syntax tests, so just doing the expr case for now gives us quite a lot of coverage already.

File: crates/compiler/test_syntax/src/test_helpers.rs (hunk 3 above)
Comment (joshuawarner32) [reply crates/compiler/test_syntax/src/test_helpers.rs:146]: (same as above; will implement this eventually - after fixing the current panics)

**Review by lukewilliamboswell — APPROVED** [review]

**Pull request merged**